### DISCUSSION TITLE
Custom Woocommerce Single Product Page

### DISCUSSION THREAD

**Discussion by digidoda** [bootscore/bootscore#578] [Q&A]
Hi there - I realise this may be a simple question, but how can I create a custom single-product template?

I see that the single product pages use woocommerce.php, but I'm still learning with bootscore and not sure where to look to get the single product page looking how I want it to so that I can pull in acf fields, shortcodes and things like that.

Any help would be much appreciated.

Mike.

**Comment by crftwrk** [comment]
You can override the WooCommerce templates in your child https://woocommerce.com/document/template-structure/. You may check https://github.com/bootscore/bootscore/issues/333 if a template is already used in theme. If template still exists in theme, use this file.

But a much more better way than template overrides is using hooks https://woocommerce.com/document/introduction-to-hooks-actions-and-filters/ in your `functions.php`. Example:

```php
add_action( 'woocommerce_before_add_to_cart_form', 'bootscore_woocommerce_before_add_to_cart_form_action' );

function bootscore_woocommerce_before_add_to_cart_form_action(){

   echo '<div class=""><p class="alert alert-info">Hook before single product cart form</p></div>';
}
```

<img width="770" alt="screenshot" src="https://github.com/bootscore/bootscore/assets/51531217/2c2db91b-a2e4-4863-9d5c-31a028887d31">


Here is a list with all available hooks https://premmerce.com/woocommerce-hooks-guide/.

And here is explained how to use this hooks with Advanced Custom Fields https://www.commercegurus.com/woocommerce-acf/.

Solved?

**Reply by digidoda** [reply]
Thanks for this Basti - if I were still to go ahead with using the custom product templates and removing the bootscore woocommerce.php file, would I need to add any files back in as my add to cart buttons now no longer work which looks like it is something to do with the add to cart params not being defined

**Reply by crftwrk** [reply]
> my add to cart buttons now no longer work which looks like it is something to do with the add to cart params not being defined

The AJAX functionality is a fine tuned area which works well with default WC settings. If you go deep into WC templates customization, there is the risk that some functions no longer work. For this reason, any function can be disabled in your child.

Disable AJAX script as described here https://bootscore.me/documentation/woocommerce/#Disable_AJAX_cart and it should work with your setup.
